### PR TITLE
collection of minor sanitisations

### PR DESCRIPTION
**Pull request by fieker**

### CODE REVIEW

File: src/Misc/Primes.jl [hunk 1]
```diff
@@ -116,6 +116,8 @@ struct PrimesSet{T}
       end
       p = next_prime(p, false)
     end
+    mod = abs(mod)
+    val = abs(val)
```
Comment (thofma) [inline src/Misc/Primes.jl:120]:
```suggestion
    val = mod(val, mod)
    val = val < 0 ? val + mod : val
```
Does this make sense? I think we allow the restriction to be "primes which are -12 modulo 5".

Comment (thofma) [inline src/Misc/Primes.jl:120]:
```suggestion
    val = mod(val, mod)
```
To make sure this works if the input values have mixed signs.

**Comment by thofma** [comment]
You can ignore the tests on 1.1.

File: src/Misc/Primes.jl [hunk 2]
```diff
@@ -117,7 +117,7 @@ struct PrimesSet{T}
       p = next_prime(p, false)
     end
     mod = abs(mod)
-    val = abs(val)
+    val = abs(val % mod)
```
Comment (thofma) [inline src/Misc/Primes.jl:120]: I don't think this is quite right.
```
julia> -1 % 3
-1
```
So this code turns -1 mod 3 into 1 mod 3.

**Comment by fieker** [comment]
On Fri, Dec 11, 2020 at 03:51:22AM -0800, thofma wrote:
> @thofma commented on this pull request.
> 
> 
> 
> > @@ -117,7 +117,7 @@ struct PrimesSet{T}
>        p = next_prime(p, false)
>      end
>      mod = abs(mod)
> -    val = abs(val)
> +    val = abs(val % mod)
> 
> I don't think this is quite right.
> ```
> julia> -1 % 3
> -1
> ```
> So this code turns -1 mod 3 into 1 mod 3.

Good thinking!

File: src/Misc/Primes.jl (hunk 2 above)
Comment (thofma) [reply src/Misc/Primes.jl:120]: mod(x, y) will always be positive if y is positive. So you can just use `val = Base.mod(val, mod)`.

**Comment by fieker** [comment]
On Fri, Dec 11, 2020 at 03:53:39AM -0800, thofma wrote:
> @thofma commented on this pull request.
> 
> 
> 
> > @@ -117,7 +117,7 @@ struct PrimesSet{T}
>        p = next_prime(p, false)
>      end
>      mod = abs(mod)
> -    val = abs(val)
> +    val = abs(val % mod)
> 
> mod(x, y) will always be positive if y is positive. So you can just use `val = Base.mod(val, mod)`.
where is this definition?

**Comment by thofma** [comment]
Docstring of mod, see `?mod`.

**Comment by fieker** [comment]
On Fri, Dec 11, 2020 at 03:58:33AM -0800, thofma wrote:
> Docstring of mod, see `?mod`. 
you win

**Pull request merged**